cbcdb/main.py

```diff
@@ -0,0 +1,375 @@
+            self.ssh_remote_bind_port = ssh_remote_bind_port if ssh_local_bind_port else self._config.ssh_remote_bind_port
+            self.ssh_local_bind_address = ssh_local_bind_address if ssh_local_bind_address else self._config.ssh_local_bind_address
+            self.ssh_local_bind_port = ssh_local_bind_port if ssh_local_bind_port else self._config.ssh_local_bind_port
+
+    def _print_debug_output(self, msg: str):
+        """
+        Prints a debug message if system is in debug mode.
+        Args:
+            msg: A string containing the message to print to the logs
+
+        Returns: None
+
+        """
+        if self._debug_mode:
+            print(f'DEBUG: {msg}')
+
+    def _get_connection(self, sql, params, method_instance):
+        """
+        Creates a connection to the database
+        Args:
+            sql: The original SQL query.
+            params: The original params.
+            method_instance: An instance of the method requesting the database connection.
+
+        Returns: The results of the original method instance.
+        """
+        conn = None
+        if self.use_ssh:
+            with open_tunnel(
+                    (self.ssh_host, self.ssh_port),
+                    ssh_username=self.ssh_user,
+                    ssh_pkey=self.ssh_key_path,
+                    remote_bind_address=(self._db_host, self.ssh_remote_bind_port),
+                    local_bind_address=(self.ssh_local_bind_address, self.ssh_local_bind_port)) as tunnel:
+                host = tunnel.local_bind_host
+                port = tunnel.local_bind_port
+                return self._database_connection_sub_method(host, port, method_instance, sql, params)
+
+        else:
+            host = self._db_host
+            port = self._db_port
+            return self._database_connection_sub_method(host, port, method_instance, sql, params)
+
+    def _database_connection_sub_method(self, host, port, method_instance, sql, params):
+        with connect(dbname=self._db_name,
+                     host=host,
+                     port=port,
+                     user=self._db_user,
+                     password=self._db_password) as conn:
+            with conn.cursor() as curs:
+                return method_instance(sql, params, curs, conn)
+
+    def get_sql_dataframe(self, sql: str, params: list = None, curs=False, conn=False) -> pd.DataFrame:
+        """
+         Returns a DataFrame for a given SQL query
+
+        Args:
+            sql: SQL string
+            params: List of parameters
+            execute_flag: If set to false, the sql query will call the connection method to open tunnel and make the
+                          database connection. The connection method will recursively call this method with the
+                          execute_flag set to true.
+
+        Returns: A Pandas Dataframe.
+
+        """
+        self._print_debug_output(f"Getting query:\n {sql}")
+
+        if curs:
```

Comment:
I see there isn't a check for `conn`, and `curs` isn't actually used in any of these calls. Should we be checking for `conn` instead? Maybe the `curs` argument can be removed?